This patch is based on reverts of upstream commits 5fba8dcb445285eb16dcec17b310742d371e5e38
and cd51d86929b169a0918bf4c4d85d984239f29806 adapted for use in the Debian package.

Index: fancy-regex/src/vm.rs
===================================================================
--- fancy-regex.orig/src/vm.rs
+++ fancy-regex/src/vm.rs
@@ -672,6 +672,7 @@ pub(crate) fn run(
 mod tests {
     use super::*;
     use quickcheck::{quickcheck, Arbitrary, Gen};
+    use rand::Rng;
 
     #[test]
     fn state_push_pop() {
@@ -786,14 +787,11 @@ mod tests {
     }
 
     impl Arbitrary for Operation {
-        fn arbitrary(g: &mut Gen) -> Self {
-            match g.choose(&[0, 1, 2]) {
-                Some(0) => Operation::Push,
-                Some(1) => Operation::Pop,
-                _ => Operation::Save(
-                    *g.choose(&[0usize, 1, 2, 3, 4]).unwrap(),
-                    usize::arbitrary(g),
-                ),
+        fn arbitrary<G: Gen>(g: &mut G) -> Self {
+            match g.gen_range(0, 3) {
+                0 => Operation::Push,
+                1 => Operation::Pop,
+                _ => Operation::Save(g.gen_range(0, 5), g.gen_range(0, usize::MAX)),
             }
         }
     }
Index: fancy-regex/Cargo.toml
===================================================================
--- fancy-regex.orig/Cargo.toml
+++ fancy-regex/Cargo.toml
@@ -32,7 +32,10 @@ version = "1.2"
 version = "0.1.8"
 
 [dev-dependencies.quickcheck]
-version = "= 1.0.1"
+version = "0.9"
+
+[dev-dependencies.rand]
+version = "0.7"
 
 [features]
 track_caller = []
